internal/home/service.go

```diff
@@ -681,3 +687,55 @@ rc_bg=YES
 
 rc_cmd $1
 `
+
+// secureBinary is used before service.Install(). This function protects AdGuardHome from
+// privilege escalation vulnerabilities caused by writable files
+func secureBinary() error {
+	switch runtime.GOOS {
+	case "windows":
+		// TODO: support windows service support securely
+		// Set file owner to admin/system and public permissions read-only
+		return errors.Error("you currently cannot install adguardhome as a service on window")
```

Comment:
This would definitely be too problematic to existing users and to Windows users in general.

I suggest instead printing a warning that explains the situation.

Reply:
The warning should probably include the link to the new [DNS KB article](https://adguard-dns.io/kb/adguard-home/running-securely/).